chore: Update ng-select to support Screen Readers

Pucek9 opened the pull request:
- update ng-select to version with better a11y support
- add aria-label for each occurences

Closes #10887

Commit:
update ng-select to v5.1.0 to add support for aira-label support

Pucek9 committed:
add aria-labels for each occurrences

saad-mo reviewed (changes requested):
`aria-label` is not updated after making a different selection from the dropdown. We would want to update it so that the screen reader announces the value that is selected vs. the initial value on load. 

<img width="1206" alt="Screen Shot 2021-04-14 at 12 55 14 PM" src="https://user-images.githubusercontent.com/15215400/114757222-34c8f980-9d21-11eb-8b21-e1b49134ef52.png">

@developpeurweb any thoughts here?

developpeurweb commented:
> `aria-label` is not updated after making a different selection from the dropdown. We would want to update it so that the screen reader announces the value that is selected vs. the initial value on load.
> 
> <img alt="Screen Shot 2021-04-14 at 12 55 14 PM" width="1206" src="https://user-images.githubusercontent.com/15215400/114757222-34c8f980-9d21-11eb-8b21-e1b49134ef52.png">
> 
> @developpeurweb any thoughts here?

@saad-mo
![Screen Shot 2021-04-14 at 3 53 19 PM](https://user-images.githubusercontent.com/15113219/114781636-ed556400-9d46-11eb-8b86-0bf270380e6a.png)

Expected behaviour for non-native components should be as per this example:
https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html
Disregard the fact that the example uses `aria-labelledby`, if you inspect and replace it with `aria-label` you will see the narration doesn't change. However, please notice the use of roles "listbox" and "option". This helps the Screen Reader make sense of divs. The idea is to shape it as close as possible to the ARIA example, especially concerning **roles, properties, states and attributes.**

This goes beyond `headings-and-labels` and overlaps with `name-role-value` but needs to be fixed.

Pucek9 committed:
Merge branch 'develop' of https://github.com/SAP/cloud-commerce-spartacus-storefront into feature/10887

Pucek9 committed:
- removed hardcoded value from placeholder
- set aria-label as value || placeholder to imitate what is currently visible (sorting component as example only)
